### PR TITLE
#5 - add UI component

### LINKED ISSUE #5: Add UI components to the project
**DOD:**

-  We have a basic UI for the website.
-  The first version have a login window and lands in a basic landing page with a welcome

### PR DESCRIPTION
**Pull request by aminations**
### **Issue #5 - Add UI component**
Add client build process, `.gitignore` for `client/`, and initial `README.md` setup

---

#### **Summary**
This PR introduces changes to integrate the `client` side of the project, adds a build step for the client, and includes project-specific configurations.

#### **Key Changes**
1. **Updated CI/CD Workflow (`deploy.yml`)**:
   - Added steps to install `client` dependencies and build the client application as part of the deployment process.
   - Ensures the `client` build process runs before server build to prepare static assets.

2. **New `.gitignore` for `client`**:
   - Created a `.gitignore` file in the `client` directory to exclude unnecessary files like:
     - `node_modules/`
     - Build outputs (`/build`)
     - Testing outputs (`/coverage`)
     - Environment variables (`.env*`)

3. **Initial Client README**:
   - Added a `README.md` for the `client` folder that provides:
     - Setup instructions for development (`npm start`).
     - Build instructions (`npm run build`).
     - Information about testing and project customization.

4. **Added `package-lock.json` for the `client`**:
   - Tracks dependencies for the client application.
   - Includes React, React-DOM, React Scripts, and TypeScript.

#### **Why This Change?**
- To integrate the frontend build process with the deployment workflow.
- To organize client-side dependencies and output files effectively.
- To document the client setup for developers.

#### **How to Test?**
1. Run the client application:
   ```bash
   cd client
   npm install
   npm start
   ```
2. Verify that the client builds successfully using:
   ```bash
   npm run build
   ```

3. Ensure the CI/CD workflow runs successfully with the updated build steps.

---

### **Checklist**
- [x] Added client build step to CI/CD pipeline.
- [x] Added a `.gitignore` file for `client`.
- [x] Created an initial `README.md` for client setup instructions.
- [x] Verified that the client app runs and builds correctly.

### CODE REVIEW

**Comment by aminations** [comment]
@rzgholizadeh, Looking at #5 , "DOD" requires a DB and login window! Should we create another issue for that?

File: client/public/images/logo-0.1.png
Comment (rzgholizadeh) [inline client/public/images/logo-0.1.png:1]: For now it is okay to put the logo image in this directory. But for future we can have a proper resource management system (for logos, images. fonts, special strings, etc)

File: client/src/App.tsx
Comment (rzgholizadeh) [inline client/src/App.tsx:1]: The next level of UI design and implementation is to do it in a modular way.
In this version there's no need for doing so. But for future development we should keep that in mind.

File: src/index.ts
Comment (rzgholizadeh) [inline src/index.ts:1]: The best practice for implementing routes for the backend or microservice is to follow the routeHandler pattern.
Please create an issue to refactor this in future, and link it to this PR so we work on it.

**Review by rzgholizadeh — APPROVED** [review]

**Comment by rzgholizadeh** [comment]
> @rzgholizadeh, Looking at #5 , "DOD" requires a DB and login window! Should we create another issue for that?

We can edit the issue or create a new one. Your call.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -74,6 +74,12 @@ jobs:
           # Clean untracked files
           git clean -fdx
 
+          # Install client dependencies and build
+          cd client
+          npm install
+          npm run build
+          cd ..
+
```
Comment (aminations) [inline .github/workflows/deploy.yml:82]: What are some strategies for updating deployment process while product is live?

File: client/public/images/logo-0.1.png
Comment (aminations) [reply client/public/images/logo-0.1.png:1]: Created Issue #16 in the backlog to work on this.

File: client/src/App.tsx
Comment (aminations) [reply client/src/App.tsx:1]: Issue #17 created in the backlog.

File: src/index.ts
Comment (aminations) [reply src/index.ts:1]: See Issue #18.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (aminations) [reply .github/workflows/deploy.yml:82]: See issue #19.

**Pull request merged**